### PR TITLE
feat: handle enum members

### PR DESCRIPTION
**Pull request by Eywek**
### All Submissions:

* [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
* [x] Have you written unit tests?
* [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
* [x] This PR is associated with an existing issue?

**Closing issues**

Closes 655

### If this is a new feature submission:

* [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

I'm not really convinced about how I've implemented it, but I've haven't found another way without duplicating or breaking code

### CODE REVIEW

File: src/metadataGeneration/typeResolver.ts [hunk 1]
```diff
@@ -425,6 +425,12 @@ export class TypeResolver {
       let referenceType: Tsoa.ReferenceType;
       if (ts.isTypeAliasDeclaration(declaration)) {
         referenceType = this.getTypeAliasReference(declaration, name, node);
+      } else if (ts.isEnumMember(declaration)) {
+        referenceType = {
+          dataType: 'refEnum',
+          refName: this.getRefTypeName(name),
+          enums: [(declaration.initializer as ts.StringLiteral).text],
```
Comment (WoH) [inline src/metadataGeneration/typeResolver.ts:432]: the declaration may not have an initializer, we when resolving an entire enum, we currently do somthing like this:

```
this.current.typeChecker.getConstantValue(declaration)
```

Comment (Eywek) [reply src/metadataGeneration/typeResolver.ts:432]: right

**Comment by WoH** [comment]
LGTM, thank you very much.

**Comment by Eywek** [comment]
closes #655

**Pull request merged**